### PR TITLE
Update README and add MagicWeather README

### PR DESCRIPTION
**Pull request by rglanz-rc**
### Checklist
- [ x ] If applicable, unit tests
- [ x ] If applicable, create follow-up issues for `purchases-ios` and hybrids

### Motivation

The main README was tweaked to be consistent with the purchases-ios README and to point to the MagicWeather example app. The MagicWeather README was written to be consistent with the iOS example app README (with Android instructions). 

### Description

The changes to the main README were mainly changes in the ordering of information or adding additional links to docs.revenuecat.com

The MagicWeather README was written using the purchases-ios README as a template. Screenshots of steps in Android Studio are included.

I pulled the most current repo and got MagicWeather working on a simulator using the instructions written.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +34,20 @@
...
+
+Or view our Android sample app:
+- [MagicWeather](examples/MagicWeather)
 
 ## Requirements
+- Java 8+
+- Minimum target: Android 4.0+ (API level 14+)
+
+## SDK Reference
+Our full SDK reference [can be found here](https://sdk.revenuecat.com/android/5.4.1/index.html).
```
Comment (aboedo) [inline README.md:50]: slight concern that this link will get outdated with every new version. 

The site is set up to redirect to the latest version, so you should be able to replace this with 
```suggestion
Our full SDK reference [can be found here](https://sdk.revenuecat.com/android/index.html).
```
and it'll be functionally the same

**Review by aboedo — CHANGES_REQUESTED** [review]
Thanks for doing this!! 

I left one comment regarding the SDK reference link, ready to go after updating that!

File: README.md [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+  * See customer transaction histories, chart lifetime value, and [grant promotional subscriptions](https://docs.revenuecat.com/docs/customers).
+  * Get notified of real-time events through [webhooks](https://docs.revenuecat.com/docs/webhooks).
+  * Send enriched purchase events to analytics and attribution tools with our easy integrations.
+
+Sign up to [get started for free](https://app.revenuecat.com/signup).
 
 ## Purchases
 
-*Purchases* is a client for the [RevenueCat](https://www.revenuecat.com/) subscription and purchase tracking system. It is an open source framework that provides a wrapper around `BillingClient` and the RevenueCat backend to make implementing in-app subscriptions in `Android` easy - receipt validation and status tracking included!
+*Purchases* is the client for the [RevenueCat](https://www.revenuecat.com/) subscription and purchase tracking system. It is an open source framework that provides a wrapper around `BillingClient` and the RevenueCat backend to make implementing in-app subscriptions in `Android` easy - receipt validation and status tracking included!
```
Comment (aboedo) [inline README.md:21]: [not needed for this PR] I'm just now realizing that we don't mention Amazon Store here, perhaps something we can incorporate in a future PR

File: examples/MagicWeather/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+Magic Weather is a sample app demonstrating the proper methods for using RevenueCat's *Purchases* SDK. This sample uses only native platform components - no third-party SDKs other than the *Purchases* SDK.
+
+Sign up for a free RevenueCat account [here](https://www.revenuecat.com).
+
+## Requirements
+
+This sample uses:
+
+- Android Studio
+- Android (API 8.0+)
```
Comment (aboedo) [inline examples/MagicWeather/README.md:12]: I believe this is actually Android 8.0+, API level 26+: 
- [minSdkVersion defined here](https://github.com/RevenueCat/purchases-android/blob/main/examples/MagicWeather/app/build.gradle#L12)
- Android version to API level mapping defined here https://developer.android.com/guide/topics/manifest/uses-sdk-element.html#ApiLevels
So I'd phrase this as either Android 8.0+ or API 26+ (or maybe both)

File: examples/MagicWeather/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+- Kotlin 1.6.21
+
+See minimum platform version requirements for RevenueCat's *Purchases* SDK [here](https://github.com/RevenueCat/purchases-android/blob/main/examples/MagicWeather/build.gradle).
+
+## Features
+
+| Feature                          | Sample Project Location                   |
```
Comment (aboedo) [inline examples/MagicWeather/README.md:19]: ❤️ this

**Review by aboedo — CHANGES_REQUESTED** [review]
One final comment and ready to go! Sorry I didn't notice it on my first pass

File: examples/MagicWeather/README.md (hunk 3 above)
Comment (rglanz-rc) [reply examples/MagicWeather/README.md]: Used both to match the Requirements section of the main readme

File: README.md (hunk 2 above)
Comment (rglanz-rc) [reply README.md:21]: I'll leave as is for now. There general summary at the top (same wording in both purchases-ios and purchases-android) would need to be changed in each repo.

Easier to do in a new PR for each.

**Review by aboedo — APPROVED** [review]
Thanks for taking care of this!!

**Pull request merged**